client/checkout/classic/upe-deferred-intent-creation/event-handlers.js

```diff
@@ -109,8 +109,45 @@ jQuery( function ( $ ) {
 		) {
 			for ( const upeElement of $( '.wcpay-upe-element' ).toArray() ) {
 				await mountStripePaymentElement( api, upeElement );
+				restrictPaymentMethodToLocation( upeElement );
 			}
 			maybeEnableStripeLink( api );
 		}
 	}
+
+	function restrictPaymentMethodToLocation( upeElement ) {
+		const paymentMethodsConfig = getUPEConfig( 'paymentMethodsConfig' );
+		const paymentMethodType = upeElement.dataset.paymentMethodType;
+		const isRestrictedInAnyCountry = !! paymentMethodsConfig[
+			paymentMethodType
+		].countries.length;
+
+		if ( isRestrictedInAnyCountry ) {
+			togglePaymentMethodForCountry(
+				paymentMethodType,
+				paymentMethodsConfig[ paymentMethodType ].countries
+			);
+			$( '#billing_country' ).on( 'change', function () {
+				togglePaymentMethodForCountry(
+					paymentMethodType,
+					paymentMethodsConfig[ paymentMethodType ].countries
+				);
+			} );
+		}
+	}
+	function togglePaymentMethodForCountry(
```

Comment:
Can you please add JS-doc so types get documented?

Reply:
Added in https://github.com/Automattic/woocommerce-payments/pull/7315/commits/6df5adeb792e62a3a2a7fbd66e85fe95ef45c0a0